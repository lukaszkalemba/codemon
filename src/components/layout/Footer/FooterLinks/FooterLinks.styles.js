import styled from 'styled-components';
import { Link as GatsbyLink } from 'gatsby';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  ${({ theme }) => theme.media.desktop`
    flex-direction: row;
    justify-content: center;
    align-items: center;
  `}
`;

const Link = styled(GatsbyLink)`
  font-size: ${({ theme }) => theme.font.size.xs};
  color: ${({ theme }) => theme.color.white};
  text-decoration: none;
  padding: 8px 0;
  opacity: 0.75;
  transition: opacity 250ms ease-in-out;

  ${({ theme }) => theme.media.desktop`
    padding: 0;
    margin-left: 80px;
    font-size: ${theme.font.size.xxs};
  `}
    :hover {
    opacity: 1;
  }
`;

export default { Wrapper, Link };
