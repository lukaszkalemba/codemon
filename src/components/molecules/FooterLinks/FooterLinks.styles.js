import styled from 'styled-components';
import { motion } from 'framer-motion';
import { Link as GatsbyLink } from 'gatsby';

const Wrapper = styled(motion.div)`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 20px 0;

  ${({ theme }) => theme.media.laptop`
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 0;
  `}
`;

const Link = styled(GatsbyLink)`
  font-size: ${({ theme }) => theme.font.size.body.s};
  color: ${({ theme }) => theme.color.white};
  text-decoration: none;
  padding: 12px 0;
  opacity: 0.8;
  transition: opacity 250ms ease-in-out, transform 250ms ease-in-out;

  ${({ theme }) => theme.media.laptop`
    padding: 0;
    margin-left: 80px;
    font-size: ${theme.font.size.body.xs};
  `}
    :hover {
    opacity: 1;
    transform: scale(1.05);
  }
`;

export default { Wrapper, Link };
