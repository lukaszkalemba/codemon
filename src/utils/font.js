export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
