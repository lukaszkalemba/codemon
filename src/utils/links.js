export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
