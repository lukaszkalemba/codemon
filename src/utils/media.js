export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
