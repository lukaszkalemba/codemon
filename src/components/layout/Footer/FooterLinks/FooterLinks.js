import React from 'react';
import links from 'utils/links';
import S from './FooterLinks.styles';

const FooterLinks = () => (
  <S.Wrapper>
    {links.map(({ id, path, name }) => (
      <S.Link key={id} to={path}>
        {name}
      </S.Link>
    ))}
  </S.Wrapper>
);

export default FooterLinks;
