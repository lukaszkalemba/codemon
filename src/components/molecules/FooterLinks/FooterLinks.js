import React, { useEffect } from 'react';
import { useInView } from 'react-intersection-observer';
import { useAnimation } from 'framer-motion';
import links from 'utils/links';
import animations from './FooterLinks.animations';
import S from './FooterLinks.styles';

const FooterLinks = () => {
  const animation = useAnimation();

  const [wrapperRef, inView] = useInView({
    triggerOnce: true,
    threshold: 0.5,
  });

  useEffect(() => {
    if (inView) {
      animation.start('animate');
    }
  }, [animation, inView]);

  const wrapperAnimations = animations.getWrapper(animation);
  return (
    <S.Wrapper ref={wrapperRef} {...wrapperAnimations}>
      {links.map(({ id, path, name }) => (
        <S.Link key={id} to={path}>
          {name}
        </S.Link>
      ))}
    </S.Wrapper>
  );
};

export default FooterLinks;
