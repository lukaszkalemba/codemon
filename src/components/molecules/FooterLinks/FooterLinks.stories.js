import React from 'react';
import FooterLinks from './FooterLinks';

export const footerLinks = () => <FooterLinks />;

export default {
  component: FooterLinks,
  title: 'Molecules|FooterLinks',
};
