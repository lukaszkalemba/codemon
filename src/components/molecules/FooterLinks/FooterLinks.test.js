import React from 'react';
import { render } from 'utils/test-utils';
import FooterLinks from './FooterLinks';

describe('<FooterLinks />', () => {
  it('renders correctly and matches snapshot', () => {
    const { container } = render(<FooterLinks />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
