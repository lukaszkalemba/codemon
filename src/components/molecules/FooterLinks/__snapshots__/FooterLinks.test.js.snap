// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterLinks /> renders correctly and matches snapshot 1`] = `
<div
  class="sc-AxjAm imGFGW"
>
  <a
    class="sc-AxirZ GNHAo"
    href="/projekty"
  >
    projekty
  </a>
  <a
    class="sc-AxirZ GNHAo"
    href="/blog"
  >
    blog
  </a>
  <a
    class="sc-AxirZ GNHAo"
    href="/wycena"
  >
    wycena
  </a>
</div>
`;
